engine/classes/ElggEntity.php

```diff
@@ -197,6 +202,19 @@ public function __set($name, $value) {
 			return;
 		}
 		if (array_key_exists($name, $this->attributes)) {
+
+			// allow string matches to integer attributes
```

Comment:
Form actions often save `"1"` to an attribute that's already `1`. In that case we don't want to record a change.